src/framework/pybind_casts.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2021.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_pybind_casts_hpp_
+#define _aer_framework_pybind_casts_hpp_
+
+#include "../simulators/stabilizer/clifford.hpp"
+
+namespace py = pybind11;
+
+namespace pybind11 {
+namespace detail {
+template <typename T> struct type_caster<matrix<T>>{
```

Comment:
I think we might also need type casters for numpy to std::vector or Vector to avoid going via a python list right?

Reply:
I think for std::vector the caster is already in place. For Vector we would need one but there wasn't any error so I guess we don't have any direct conversion from input to Vector. I'll check, because sometimes it's confusing working with pybind11.

Reply:
Iirc think currently all inputs are converted to `std::vector` anyway, `AER::Vector` is only used in results